experiment_prototype/scan_classes/sequences.py

```diff
@@ -311,10 +229,13 @@ def __init__(self, seqn_keys, sequence_slice_dict, sequence_interface, transmit_
         # a specific rxrate (given by the config).
 
         # number of samples for the first range for all slice ids
-        first_range_samples = {slice_id : int(math.ceil(self.slice_dict[slice_id]['first_range']/self.slice_dict[slice_id]['range_sep']))
-            for slice_id in self.slice_ids}
 
-        # time for number of ranges given, in us, taking into account first_range and num_ranges.          
+        range_as_samples = lambda x,y: int(math.ceil(x/y))
+        first_range_samples = {slice_id : range_as_samples(self.slice_dict[slice_id]['first_range'],
+                                                            self.slice_dict[slice_id]['range_sep'])
```

Comment:
thinking about this a bit more, I think ssdelay could be incorrect also if pulse_len != 1/rx_rate because that's an assumption for num_ranges as well. we might need to factor pulse_len into the ssdelay calculation

Reply:
I will make a small PR to this branch for some updates to these calcs for numberofreceivesamples

Reply:
Good catch. I think that the finding these ways to simplify the code will allow us to highlight more potential issues.

Reply:
just fyi this PR is open with a fix for this #177